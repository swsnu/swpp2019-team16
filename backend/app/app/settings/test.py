from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
