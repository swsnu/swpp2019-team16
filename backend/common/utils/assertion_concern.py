class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
