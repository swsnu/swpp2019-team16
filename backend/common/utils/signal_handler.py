import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
