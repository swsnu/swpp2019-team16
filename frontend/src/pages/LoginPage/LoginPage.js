import React from 'react';
import Main from '../_layouts/Main';
import LoginContainer from '../../containers/LoginContainer/LoginContainer';

function LoginPage() {
  return (
    <Main>
      <LoginContainer />
    </Main>
  );
}
export default LoginPage;
