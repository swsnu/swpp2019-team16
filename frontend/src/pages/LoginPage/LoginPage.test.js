import React from 'react';
import LoginPage from './LoginPage';
import { MemoryRouter, Route } from 'react-router-dom';
import { renderWithRedux } from 'test/utils';

describe('<LoginPage />', () => {
  const initialState = {
    auth: {
      login: {
        email: '',
        password: '',
      },
      auth: null,
    }
  };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/login']}>
        <Route component={LoginPage} path="/login" />
      </MemoryRouter>,
      initialState,
    );
    expect(container).toMatchSnapshot();
  });
});
