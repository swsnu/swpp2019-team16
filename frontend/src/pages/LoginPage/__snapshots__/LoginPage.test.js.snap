// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bPGLpq"
  >
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h4"
        >
          Ya-Ta!
        </h4>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-38 makeStyles-root-41 MuiButton-text makeStyles-text-40 makeStyles-text-43 MuiButton-text makeStyles-text-40 makeStyles-text-43"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Log In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="sc-bwzfXH uTxCW"
    >
      <div
        class="MuiBox-root MuiBox-root-76"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          Login to Ya-Ta!
        </h3>
        <div
          class="MuiBox-root MuiBox-root-77"
        >
          <input
            id="email-input"
            name="email"
            placeholder="email"
            type="text"
            value=""
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root-78"
        >
          <input
            id="password-input"
            name="password"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root-79"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-38 makeStyles-root-80 MuiButton-contained makeStyles-contained-39 makeStyles-contained-81 MuiButton-contained makeStyles-contained-39 makeStyles-contained-81"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Login
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-38 makeStyles-root-83 MuiButton-contained makeStyles-contained-39 makeStyles-contained-84 MuiButton-contained makeStyles-contained-39 makeStyles-contained-84"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Register
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
